apps/dashboard/app/models/batch_connect/app.rb

```diff
@@ -135,7 +135,9 @@ def clusters
     # Whether this is a valid app the user can use
     # @return [Boolean] whether valid app
     def valid?
-      form_config.any? && configured_clusters.any? && clusters.any?
+      form_config.any? &&
+        # top level cluster(s) is defined and valid OR form.cluster exists
+        ((configured_clusters.any? && clusters.any?) || form_config.fetch(:form, []).include?('cluster'))
```

Comment:
I'd prefer:

```ruby
def valid?
  if form_config.empty?
    false
  elsif configured_clusters.any?
    clusters.any?
  else
    true
  end
end
```

So if there are no configured clusters but is a form config then we don't flag it as invalid.

Reply:
Updated. At this time, users with no cluster in the `form.yml` would get a nil exception while trying to access `context.cluster` at some point in submission, but that should be fixed in #529.